dataeng/processed_data/process_data.py

```diff
@@ -0,0 +1,208 @@
+import csv
+import os
+import time
+
+
+class processing:
+
+    def __init__(self, source_data_path: str, processed_data_path: str):
+        """
+        Source_data_path is the path where source data are located
+        """
+        self.source_data_path = source_data_path
+        self.processed_data_path = processed_data_path
+
+    def get_age(self, birth_date: float):
+        birth_date = float(birth_date)
+        current_time = time.time() * 1000
+        return (current_time - birth_date) / (3.154 * 10 ** 10)
+
+
+    def filter_data(self, is_image_exists: int, min_age: float, max_age: float):
+        """
+        Filter_data applies for data after processing (output.csv)
+        is_image_exists: 1 => return users that have images, 0 => they don't have, -1 => return both
+        min_age: return users that their ages in years is higher than min age, input is float or -1 if we don't want to apply this filter
+        max_age: return users that their ages in years is less than max age, input is float or -1 if we don't want to apply this filter
+
+        this filter also provide the possibility to apply filters together (Ex: users who don't have images and thier age is between 20 and 40)
+        Also it is possible to apply exactly two filters (Ex: users who have images and thier min age is 30)
+        """
+        min_age = float(min_age)
+        max_age = float(max_age)
+        records = []
+
+        out_file_name = 'output.csv'
+        out_file = f'{self.processed_data_path}/{out_file_name}'
+
+        with open(out_file, 'r') as file:
+                csvreader = csv.reader(file)
+                next(csvreader, None)
+                for record in csvreader:
+                    if len(record) != 0:
+                        records.append(record)
+        file.close()
+
```

Comment:
There is no need to close the file explicitly. It will be closed automatically right after the end of **with** clause